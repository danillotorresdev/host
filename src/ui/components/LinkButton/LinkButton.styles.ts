import { styled } from "@linaria/react";
import { Link } from "react-router-dom";

export const LinkButton = styled(Link)`
  display: flex;
  align-items: center;
  justify-content: center;
  height: 40px;
  font-size: 14px;
  padding: 0 0.5rem;
  border: none;
  border-radius: 0.5rem;
  cursor: pointer;
  transition: background-color 0.3s ease;
  background-color: transparent;
  color: #000;
  text-decoration: none;
  appearance: none;
  border: 1px solid #000;
  width: 100%;

  &:hover {
    color: #000;
  }

  &:disabled {
    opacity: 0.6;
    cursor: not-allowed;
  }
`;
