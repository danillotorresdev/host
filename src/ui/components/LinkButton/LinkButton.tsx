import { LinkProps } from "react-router-dom";
import * as S from "./LinkButton.styles";

export const LinkButton = ({ children, to, ...props }: Readonly<LinkProps>) => (
  <S.LinkButton to={to} {...props}>
    {children}
  </S.LinkButton>
);
