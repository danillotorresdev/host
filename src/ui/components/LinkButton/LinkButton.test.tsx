import { LinkButton } from "./LinkButton";
import { render } from "@/utils/settings/tests/utilities";
import { BrowserRouter } from "react-router-dom";
import { expect, it, describe } from "vitest";

describe("LinkButton", () => {
  it("should render the user's name", () => {
    const { asFragment } = render(
      <LinkButton to="/user/jhondoe/repos">
        Reposit√≥rios
      </LinkButton>,
      {
        wrapper: BrowserRouter,
      }
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
