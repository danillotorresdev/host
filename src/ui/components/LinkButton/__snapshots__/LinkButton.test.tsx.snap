// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LinkButton > should render the user's name 1`] = `
<DocumentFragment>
  <a
    class="mocked-styled-0 ltiwt9s"
    href="/user/jhondoe/repos"
  >
    Reposit√≥rios
  </a>
</DocumentFragment>
`;
